python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -186,14 +184,117 @@ def _makeRecords(dataIds, universe):
             for dimension, values in expandedIds.items()}
 
 
-def expandUniqueId(butler, partialId):
+def _fillAllKeys(dimension: Dimension, value: Any) -> Mapping[str, Any]:
+    """Create an arbitrary mapping of all required keys for a given dimension
+    that do not refer to other dimensions.
+
+    Parameters
+    ----------
+    dimension : `lsst.daf.butler.Dimension`
+        The dimension for which to generate a set of keys (e.g., detector).
+    value
+        The value assigned to ``dimension`` (e.g., detector ID).
+
+    Returns
+    -------
+    expandedValue : `dict` [`str`]
+        A mapping of dimension keys to values. ``dimension's`` primary key
+        maps to ``value``, but all other mappings (e.g., detector name)
+        are arbitrary.
+    """
+    expandedValue = {}
+    for key in dimension.uniqueKeys:
+        if key.nbytes:
+            castType = bytes
+        else:
+            castType = key.dtype().python_type
+        try:
+            castValue = castType(value)
+        except TypeError:
+            castValue = castType()
+        expandedValue[key.name] = castValue
+    for key in dimension.metadata:
+        if not key.nullable:
+            expandedValue[key.name] = key.dtype().python_type(value)
+    return expandedValue
+
+
+def _matchAnyDataId(record: Mapping[str, Any], registry: Registry, dimension: Dimension):
+    """Matches a partial dimension record to an existing record along a
+    specific dimension.
```

Comment:
I'm a little surprised this functionality, as well as `_fillRelationships`, doesn't already exist somewhere you could import from elsewhere in daf_butler. (Though I don't know immediately where - perhaps it's tied up with way more than is needed for a test repo?).

Reply:
The problem is that in a real repo you don't want to do this; you want your detectors, exposures, visits, tracts, etc. to be dictated by real data. All these functions basically make up filler values.

Reply:
Ah, interesting. Consider adding something akin to, "Only for creating filler dimension data in a test repo, as datasets in a real repo have known dimension values." Otherwise I might be tempted to "borrow" functions like these to complete/tidy up/fill gaps in a real repo.